raiden/network/transport/matrix.py

```diff
@@ -297,10 +296,7 @@ def send_async(
             queue_identifier=queue_identifier,
         )
 
-        if isinstance(message, Processed):
-            self._send_immediate(queue_identifier, message)
-        else:
-            self._send_with_retry(queue_identifier, message)
+        self._send_with_retry(queue_identifier, message)
```

Comment:
https://github.com/raiden-network/raiden/pull/2268#discussion_r213388435 ?

Reply:
@hackaugusto Yes, exactly. After you explained that the `Delivered` would go in different queues, and therefore `Processed` should also be retried, I agreed with your comment. This was made clear in this issue, as the `Processed` wasn't being cleared (but wasn't being retried either, only in the next session). This commit fixes it.